python/lsst/afw/table/aggregates.cc

```diff
@@ -21,25 +21,185 @@
  */
 
 #include <pybind11/pybind11.h>
-//#include <pybind11/operators.h>
-//#include <pybind11/stl.h>
+#include <pybind11/operators.h>
+#include <pybind11/stl.h>
+
+#include "numpy/arrayobject.h"
+#include "ndarray/pybind11.h"
+#include "ndarray/converter.h"
+
+#include "lsst/afw/geom/ellipses/Quadrupole.h"
+
+#include "lsst/afw/coord/Coord.h"
+#include "lsst/afw/geom/Angle.h"
+#include "lsst/afw/table/Key.h"
+#include "lsst/afw/table/Schema.h"
+#include "lsst/afw/table/BaseRecord.h"
+#include "lsst/afw/table/FunctorKey.h"
+#include "lsst/afw/table/aggregates.h"
 
 namespace py = pybind11;
+using namespace pybind11::literals;
 
 using namespace lsst::afw::table;
 
+template <typename T>
+void declarePointKey(py::module &mod, const std::string & suffix){
+    py::class_<PointKey<T>> clsPointKey(mod, ("Point"+suffix+"Key").c_str());
+    /* Constructors */
+    clsPointKey.def(py::init<>());
+    clsPointKey.def(py::init<Key<T> const &, Key<T> const &>());
+    clsPointKey.def(py::init<SubSchema const &>());
+    /* Operators */
+    clsPointKey.def(py::self == py::self);
+    clsPointKey.def(py::self != py::self);
+    /* Members */
+    clsPointKey.def("getX", &PointKey<T>::getX);
+    clsPointKey.def("getY", &PointKey<T>::getY);
+    clsPointKey.def("isValid", &PointKey<T>::isValid);
+    clsPointKey.def_static("addFields", &PointKey<T>::addFields);
+    clsPointKey.def("set", [](PointKey<T> & pointKey, BaseRecord & record, lsst::afw::geom::Point<T,2> const & value){
+        return pointKey.set(record, value);
+    });
+    clsPointKey.def("get", &PointKey<T>::get);
+};
+
+template <typename T, int N>
+void declareCovarianceMatrixKey(py::module &mod, const::std::string & suffix){
+    typedef std::vector< Key<T> > SigmaKeyArray;
+    typedef std::vector< Key<T> > CovarianceKeyArray;
+    typedef std::vector<std::string> NameArray;
+    
+    py::class_<CovarianceMatrixKey<T,N>> cls(mod, ("CovarianceMatrix"+suffix+"Key").c_str());
+    
+    cls.def(py::init<>());
+    cls.def(py::init<SigmaKeyArray const &, CovarianceKeyArray const &>(),
+            "sigma"_a, "cov"_a=CovarianceKeyArray());
+    cls.def(py::init<SubSchema const &, NameArray const &>());
+    
+    cls.def_static("addFields", (CovarianceMatrixKey<T,N> (*)(
```

Comment:
``CovarianceMatrixKey<T,N>`` is the return type, and the ``(*)`` is needed to cast the function. Typically this would be ``(CovarianceMatrixKey<T,N>::*)``, but since this is a static method we drop the ``CovarianceMatrixKey<T,N>::``. This is required to wrap overloaded functions.